### PR TITLE
Revamp table BBEs

### LINKED ISSUE #3777: Improve Table related BBEs
$title

### PR DESCRIPTION
**Pull request by ushirask**

### CODE REVIEW

File: examples/multiple-key-fields/multiple_key_fields.bal [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
+   int salary;
 };
 
 public function main() {
-    // `t` has a key sequence with `firstName` and `lastName` fields.
-    table<Employee> key(firstName, lastName) t = table [
-        {firstName: "John", lastName: "Smith", salary: 100},
-        {firstName: "Fred", lastName: "Bloggs", salary: 200}
-    ];
+   // `employees` has a key sequence with `firstName` and `lastName` fields.
```
Comment (praneesha) [inline examples/multiple-key-fields/multiple_key_fields.bal:10]:
```suggestion
   // `employees` has a key sequence with the `firstName` and `lastName` fields.
```

File: examples/multiple-key-fields/multiple_key_fields.bal [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
+   ];
 
-    // The key sequence provides keyed access to members of the `table`.
-    Employee? e = t["Fred", "Bloggs"];
+   // The key sequence provides keyed access to members of the `table`.
+   Employee? e = employees["John", "Bloggs"];
+   io:println(e);
 
-    io:println(e);
+   // `employees` has a key sequence with `firstName` and `lastName` fields.
```
Comment (praneesha) [inline examples/multiple-key-fields/multiple_key_fields.bal:20]:
```suggestion
   // `employees` has a key sequence with the `firstName` and `lastName` fields.
```

File: examples/multiple-key-fields/multiple_key_fields.md [hunk 3]
```diff
@@ -1,6 +1,10 @@
 # Multiple key fields
 
-A `table` provides access to its members using a key that comes from the `read-only` fields of the member. It is a key sequence, which is used to provide keyed access to its members. The key sequence is an ordered sequence of field names.
+A `table` provides access to its members using a key that comes from the `readonly` fields of the member. It is a key sequence, which is used to provide keyed access to its members. The key sequence is an ordered sequence of field names.
```
Comment (praneesha) [inline examples/multiple-key-fields/multiple_key_fields.md:3]:
```suggestion
A `table` provides access to its members using a key that comes from the `readonly` fields of the member. It is a key sequence, which is used to provide keyed access to its members. The key sequence is an ordered sequence of the field names.
```

File: examples/multiple-key-fields/multiple_key_fields.metatags [hunk 4]
```diff
@@ -1,2 +1,2 @@
 description: This BBE demonstrates multiple key fields in tables in Ballerina.
```
Comment (praneesha) [inline examples/multiple-key-fields/multiple_key_fields.metatags:1]:
```suggestion
description: This BBE demonstrates multiple key fields in tables of Ballerina.
```

File: examples/structured-keys/structured_keys.bal [hunk 5]
```diff
@@ -1,20 +1,25 @@
...
+   int salary;
 };
 
 public function main() {
-    // key field, `name` is of `record` type.
-    table<Employee> key(name) t = table [
-        {name: {first: "John", last: "Smith"}, salary: 100},
-        {name: {first: "Fred", last: "Bloggs"}, salary: 200}
-    ];
+   // key field, `name` is of `record` type.
```
Comment (praneesha) [inline examples/structured-keys/structured_keys.bal:12]:
```suggestion
   //In the key field, the `name` is of the `record` type.
```

File: examples/structured-keys/structured_keys.bal [hunk 6]
```diff
@@ -1,20 +1,25 @@
...
+       {name: {first: "Fred", last: "Bloggs"}, salary: 200}
+   ];
 
-    Employee? e = t[{first: "Fred", last: "Bloggs"}];
-    io:println(e);
+   Employee? e = t[{first: "Fred", last: "Bloggs"}];
+   io:println(e);
+
+   record {|string first; string last;|} n = {first: "Sam", last: "Smith"};
+   // make the key immutable using `cloneReadOnly()`
```
Comment (praneesha) [inline examples/structured-keys/structured_keys.bal:22]:
```suggestion
   // Make the key immutable using `cloneReadOnly()`.
```

File: examples/structured-keys/structured_keys.md [hunk 7]
```diff
@@ -1,6 +1,13 @@
 # Structured keys
 
-Key fields can be structured as long as they belong to any subtype of plain data. The value of the key field must be immutable. The initializer of the `readonly` field will be constructed as immutable. In other cases, `cloneReadOnly()` can be used to create an immutable value.
+Key fields can be structured types as long as they belong to any subtype of plain data. Plain data is a simple value, a sequence value or a structured value where all members are plain data.
```
Comment (praneesha) [inline examples/structured-keys/structured_keys.md:3]:
```suggestion
Key fields can be structured types as long as they belong to any subtype of plain data. Plain data is a simple value, a sequence value, or a structured value where all members are plain data.
```

File: examples/structured-keys/structured_keys.md [hunk 8]
```diff
@@ -1,6 +1,13 @@
...
+The value of the key field must be immutable.
+The initializer of the readonly field will be constructed as immutable. When using `add`/`put` cloneReadOnly() can be used to create an immutable value.
```
Comment (praneesha) [inline examples/structured-keys/structured_keys.md:5]:
```suggestion
The initializer of the read-only field will be constructed as immutable. When using `add`/`put`, `cloneReadOnly()` can be used to create an immutable value.
```

File: examples/structured-keys/structured_keys.metatags [hunk 9]
```diff
@@ -1,2 +1,2 @@
 description: This BBE demonstrates structured keys in tables in Ballerina.
```
Comment (praneesha) [inline examples/structured-keys/structured_keys.metatags:1]:
```suggestion
description: This BBE demonstrates structured keys in tables of Ballerina.
```

**Review by lasinicl — COMMENTED** [review]
It seems the ballerina code is not properly formatted. Can you check?

File: examples/multiple-key-fields/multiple_key_fields.md [hunk 10]
```diff
@@ -1,6 +1,10 @@
 # Multiple key fields
 
-A `table` provides access to its members using a key that comes from the `read-only` fields of the member. It is a key sequence, which is used to provide keyed access to its members. The key sequence is an ordered sequence of field names.
+A `table` provides access to its members using a key that comes from the `readonly` fields of the member. It is a key sequence, which is used to provide keyed access to its members. The key sequence is an ordered sequence of the field names.
+
+## Related links
+- [Table Syntax](/learn/by-example/table-syntax/)
+- [Structured keys](/learn/by-example/multiple-key-fields/)
```
Comment (lasinicl) [inline examples/multiple-key-fields/multiple_key_fields.md:7]: May be we can add `Maps` BBE as a related link as the table rows are mappings :thinking:

File: examples/structured-keys/structured_keys.md [hunk 11]
```diff
@@ -1,6 +1,13 @@
 # Structured keys
 
-Key fields can be structured as long as they belong to any subtype of plain data. The value of the key field must be immutable. The initializer of the `readonly` field will be constructed as immutable. In other cases, `cloneReadOnly()` can be used to create an immutable value.
+Key fields can be structured types as long as they belong to any subtype of plain data. Plain data is a simple value, a sequence value, or a structured value where all members are plain data.
+The value of the key field must be immutable.
+The initializer of the read-only field will be constructed as immutable. When using `add`/`put`, `cloneReadOnly()` can be used to create an immutable value.
```
Comment (lasinicl) [inline examples/structured-keys/structured_keys.md:5]: Shall we add the description in a single line? The contents in separate lines are rendered as separate paragraphs on the website.

File: examples/structured-keys/structured_keys.out [hunk 12]
```diff
@@ -1,2 +1,3 @@
 $ bal run structured_keys.bal
 {"name":{"first":"Fred","last":"Bloggs"},"salary":200}
+[{"name":{"first":"John","last":"Smith"},"salary":100},{"name":{"first":"Fred","last":"Bloggs"},"salary":200},{"name":{"first":"Sam","last":"Smith"},"salary":23}]
```
Comment (lasinicl) [inline examples/structured-keys/structured_keys.out:3]: Let's add a new line at the end of the file

**Review by lasinicl — APPROVED** [review]

**Review by KavinduZoysa — APPROVED** [review]

**Pull request merged**